esphome/core/helpers.cpp

```diff
@@ -258,6 +258,39 @@ optional<int> parse_int(const std::string &str) {
     return {};
   return value;
 }
+
+optional<int> parse_hex(const char chr) {
+  int out = chr;
+  if (out >= '0' && out <= '9')
+    return (out - '0');
+  if (out >= 'A' && out <= 'F')
+    return (10 + (out - 'A'));
+  if (out >= 'a' && out <= 'f')
+    return (10 + (out - 'a'));
+  return {};
+}
+
+optional<int> parse_hex(const std::string &str, size_t start, size_t length) {
+  if (str.length() < start) {
+    return {};
+  }
+  size_t len = start + length;
+  if (str.length() < len) {
+    return {};
+  }
+  int out = 0;
+  for (size_t i = start; i < len; i++) {
+    char chr = str[i];
```

Comment:
Just one thing on the variable name, but not a show stopper =)
```suggestion
  size_t end = start + length;
  if (str.length() < end) {
    return {};
  }
  int out = 0;
  for (size_t i = start; i < end; i++) {
    char chr = str[i];
```